src/m365/planner/commands/bucket/bucket-get.ts

```diff
@@ -0,0 +1,215 @@
+import { PlannerBucket } from '@microsoft/microsoft-graph-types';
+import { Logger } from '../../../../cli';
+import { AxiosRequestConfig } from 'axios';
+import {
+  CommandOption
+} from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import { validation } from '../../../../utils';
+import GraphCommand from '../../../base/GraphCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  id?: string;
+  title?: string;
+  planId?: string;
+  planName?: string;
+  ownerGroupId?: string;
+  ownerGroupName?: string;
+}
+
+class PlannerBucketGetCommand extends GraphCommand {
+  public get name(): string {
+    return commands.BUCKET_GET;
+  }
+
+  public get description(): string {
+    return 'Gets the Microsoft Planner bucket in a plan';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.id = typeof args.options.id !== 'undefined';
+    telemetryProps.title = typeof args.options.title !== 'undefined';
+    telemetryProps.planId = typeof args.options.planId !== 'undefined';
+    telemetryProps.planName = typeof args.options.planName !== 'undefined';
+    telemetryProps.ownerGroupId = typeof args.options.ownerGroupId !== 'undefined';
+    telemetryProps.ownerGroupName = typeof args.options.ownerGroupName !== 'undefined';
+    return telemetryProps;
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['id', 'name', 'planId', 'orderHint'];
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    this
+      .getBucketId(args)
+      .then((bucketId: string) => this.getBucketById(bucketId))
+      .then((bucket: PlannerBucket) => {
+        logger.log(bucket);
+        cb();
+      }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+  }
+
+  private getBucketId (args: CommandArgs): Promise<string> {
+    if (args.options.id) {
+      return Promise.resolve(args.options.id);
+    }
+
+    return this
+      .getPlanId(args)
+      .then((planId: string) => {
+        const requestOptions: any = {
+          url: `${this.resource}/v1.0/planner/plans/${planId}/buckets`,
+          headers: {
+            accept: 'application/json;odata.metadata=none'
+          },
+          responseType: 'json'
+        };
+
+        return request.get<{ value: { id: string; name: string; }[] }>(requestOptions);
+      })
+      .then(response => {
+        const bucket: { id: string; name: string; } | undefined = response.value.find(val => val.name === args.options.title);
```

Comment:
We should also handle the case if we find multiple buckets with the same name. You find a case here: https://github.com/pnp/cli-microsoft365/blob/f028e14fa587213b1868cdee83f77b7c99d109ce/src/m365/planner/commands/bucket/bucket-remove.ts#L114-L124